rockcraft/models/project.py

```diff
@@ -186,9 +182,12 @@ def _validate_all_platforms(cls, platforms: dict[str, Any]) -> dict[str, Any]:
             # Make sure the provided platform_set is valid
             try:
                 platform = Platform(**platform).dict()
-            except CraftValidationError as err:
-                # pylint: disable=raise-missing-from
-                raise CraftValidationError(f"{error_prefix}: {str(err)}")
+            except pydantic.ValidationError as err:
+                errors = []
+                for err_dict in err.errors():
+                    errors.append(err_dict["msg"])
```

Comment:
You can even use it in join() directly
```suggestion
                errors = [ err_dict["msg"] for err_dict in err.errors()] 
```